src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1219 @@
+                    # divisibility for
+                    sub_poly = current_poly.subs(
+                        **{str(self._names[j]): exchange_polys[j]})
+
+                    # calculate maximal power of exchange_polys[j] that
+                    # divides sub_poly
+                    counter = 0
+                    while True:
+                        (q, r) = sub_poly.quo_rem(
+                            exchange_polys[j] ** (counter + 1))
+                        if r != 0:
+                            break
+                        counter = counter + 1
+
+                    # divide exchange polynomial by this maximal power
+                    laurent_polys[i] = laurent_polys[i] / (
+                        self._polynomial_ring(self._names[j]) ** counter)
+
+        # after performing all substitutions, set internal laurent polynomials
+        self._laurent_polys = laurent_polys
+
+    # mutates self at ith variable
+
+    def mutate(self, i, inplace=True):
+        r"""
+        Mutate this seed at the ``i`` th index.
+
+        INPUT:
+
+        - ``i`` -- integer or iterable of integers; the index/indices to mutate
+          ``self`` at, where we index from 0.
+
+        - ``inplace`` -- boolean (default: ``True``); whether to mutate the
+          current instance of the seed, or return a new ``LpaSeed`` object.
+
+        EXAMPLES:
+
+        We mutate a rank-two Laurent phenomenon algebra at the first index::
+
+            sage: var('x1,x2,a0,a2,b0,b1')
+            (x1, x2, a0, a2, b0, b1)
+            sage: S = LpaSeed({x1: a0 + a2*x2, x2: b0 + b1*x1}, coefficients=[a0,a2,b0,b1])
+            sage: print(S)
+            A seed with cluster variables [x1, x2] and exchange polynomials [x2*a2 + a0, x1*b1 + b0]
+            sage: S.mutate(0); print(S)
+            A seed with cluster variables [(x2*a2 + a0)/x1, x2] and exchange polynomials [x2*a2 + a0, x1*b0 + a0*b1]
+
+        Mutating at the same index is an involution::
+
+            sage: var('x1,x2,x3')
+            (x1, x2, x3)
+            sage: S = LpaSeed({x1: 1 + x2*x3, x2: 1 + x1^2 + x3^2, x3: 1 + x1 + x2})
+            sage: T = S.mutate([2,2], inplace=False)
+            sage: print(T == S)
+            True
+        """
+
+        # input preprocessing
+
+        if not isinstance(inplace, bool):
+            raise TypeError('The parameter inplace must be a boolean; got %s'
+                            % (type(inplace)))
+
+        if not inplace:
+            self = LpaSeed(self)
+
+        # is our input data iterable?
+        if hasattr(i, '__iter__'):
+            for index in i:
+                if index not in range(0, self._rank):
+                    raise IndexError('Iterable %s contains a nonvalid index %s'
+                                     % (i, index))
+                self.mutate(index)
+            if not inplace:
+                return self
+            else:
+                return
```

Comment:
```suggestion
            return self
```
It is actually surprising useful to return the object, even when mutating inplace. It makes chaining operations much easier. (Analogous changes needed below.)